src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java

```diff
@@ -139,7 +139,7 @@ public final void setIllegalPkgs(String... from) {
         illegalPkgs = from.clone();
         illegalPkgsRegexps.clear();
         for (String illegalPkg : illegalPkgs) {
-            illegalPkgsRegexps.add(CommonUtil.createPattern("^" + illegalPkg + "\\..*"));
+            illegalPkgsRegexps.add(CommonUtil.createPattern(illegalPkg + "\\..*"));
```

Comment:
Won't this possibly create false matches like if my illegal package is `blah` and my input file has a package `com.other.blah.other`?

Reply:
@rnveach This will cause an error when the blah is found in the package, and this is an illegal package

Reply:
I am not sure I follow what you mean by an error. I assume violation.

But my question was would this be a false match. If I say my illegal package is `blah`, I expect to only match `blah` or `blah.other` depending on the rest of the criteria for this module. I wouldn't expect a match on `com.other.blah.other`.

Reply:
> I am not sure I follow what you mean by an error. I assume violation.
> 
> But my question was would this be a false match. If I say my illegal package is `blah`, I expect to only match `blah` or `blah.other` depending on the rest of the criteria for this module. I wouldn't expect a match on `com.other.blah.other`.

Thanks @rnveach In this case it will give false violation i removed the ^ symbol, can we add some conditions like only static or some more keywords allow only

Reply:
You would need to examine the module closer to see if there is some conditional statements that would have to be added. Like other issues, I have not looked into the full details of the issue and code.

However, to me it does seem odd that this issue can be resolved with what amounts to a configuration change from looking at this PR. If your changing the code in the property setter, your basically saying either how we read the value from the configuration, or the contents of the configuration, are wrong.

Reply:
@romani @rnveach There are the proper usage how to use import in different, For this issue, We need a condition only for handling the static instead of removing the "^" from the import check.

- Single Type Import:

This is the most common usage of the import statement and is used to import a single class or interface.

`import java.util.ArrayList;`
Import All Classes from a Package:
You can import all classes and interfaces from a specific package by using the * wildcard.


`import java.util.*;`
This will import all classes and interfaces within the java.util package.

- Static Import:

The import statement can also be used to import static members (fields and methods) of a class. This is often used for constants or utility methods.


`import static java.lang.Math.*;`
This allows you to use static members of the Math class directly without prefixing with Math..

- On-Demand Type Import:

This is similar to importing all classes from a package but only imports the specific class you use in your code.


```
import java.util.ArrayList;
import java.util.HashMap;
```
In this example, only ArrayList and HashMap are imported from the java.util package, even though other classes and interfaces are available in that package.

- Import for Nested Classes:

If you have a nested class, you can import it in a similar way, using the nested class name.


`import mypackage.OuterClass.InnerClass;`
This imports the InnerClass which is a nested class within OuterClass.

Reply:
> java.util.HashMap
> mypackage.OuterClass.InnerClass

Just pointing out that you can't identify an inner class from a regular class without multi-file and classpath support.

Reply:
@rnveach @romani can we add separate property for this for handling static import?

Reply:
It is completely impossible for checkstyle to cover it, so we simply need to skip such cases from validation.